pkg/manifest/manifest.go

```diff
@@ -98,6 +110,69 @@ func (m *Manifest) UnmarshalJSON(in []byte) error {
 	return validateResourceId(m.id)
 }
 
+func (m *Manifest) Include(excludeIdentifier string, includeTechPreview bool, profile string, capabilities ClusterCapabilities) error {
+
+	annotations := m.Obj.GetAnnotations()
+	if annotations == nil {
+		return fmt.Errorf("no annotations")
+	}
+
+	excludeAnnotation := fmt.Sprintf("exclude.release.openshift.io/%s", excludeIdentifier)
+	if v := annotations[excludeAnnotation]; v == "true" {
+		return fmt.Errorf("%s=%s", excludeAnnotation, v)
+	}
+
+	featureGateAnnotation := "release.openshift.io/feature-gate"
+	featureGateAnnotationValue, featureGateAnnotationExists := annotations[featureGateAnnotation]
+	if featureGateAnnotationValue == "TechPreviewNoUpgrade" && !includeTechPreview {
+		return fmt.Errorf("tech-preview excluded, and %s=%s", featureGateAnnotation, featureGateAnnotationValue)
+	}
+	// never include the manifest if the feature-gate annotation is outside of allowed values (only TechPreviewNoUpgrade is currently allowed)
+	if featureGateAnnotationExists && featureGateAnnotationValue != "TechPreviewNoUpgrade" {
+		return fmt.Errorf("unrecognized value %s=%s", featureGateAnnotation, featureGateAnnotationValue)
+	}
+
+	profileAnnotation := fmt.Sprintf("include.release.openshift.io/%s", profile)
+	if val, ok := annotations[profileAnnotation]; ok && val != "true" {
+		return fmt.Errorf("unrecognized value %s=%s", profileAnnotation, val)
+	} else if !ok {
+		return fmt.Errorf("%s unset", profileAnnotation)
+	}
+
+	return CheckResourceEnablement(annotations, capabilities)
+}
+
+// CheckResourceEnablement, given resource annotations and defined cluster capabilities, checks if the capability
+// annotation exists. If so, each capability name is validated against the known set of capabilities. Each valid
+// capability is then checked if it is disabled. If any invalid capabilities are found an error is returned listing
+// all invalid capabilities. Otherwise, if any disabled capabilities are found an error is returned listing all
+// disabled capabilities.
+func CheckResourceEnablement(annotations map[string]string, capabilities ClusterCapabilities) error {
```

Comment:
This does not need to be part of the public API (the public aspect is `Include`, this is just an internal helper).  Can we rename to `checkResourceEnablement`?

Reply:
yes, will fix it

Reply:
Fixed in https://github.com/openshift/library-go/commit/d2d9b0ce7cccf99dfd5903d246b0e619dc764096